spark/src/main/java/org/apache/iceberg/spark/SparkUtil.java

```diff
@@ -61,4 +65,42 @@ public static void validatePartitionTransforms(PartitionSpec spec) {
           String.format("Cannot write using unsupported transforms: %s", unsupported));
     }
   }
+
+  /**
+   * A modified version of Spark's LookupCatalog.CatalogAndIdentifier.unapply
+   * Attempts to find the catalog and identifier a multipart identifier represents
+   * @param nameParts Multipart identifier representing a table
+   * @return The CatalogPlugin and Identifier for the table
+   */
+  public static <C, T> Pair<C, T> catalogAndIdentifier(List<String> nameParts,
+                                                       Function<String, C> catalogProvider,
+                                                       IdentiferFunction<T> identiferProvider,
+                                                       C defaultCatalog,
+                                                       String[] currentNamespace) {
+    Preconditions.checkArgument(!nameParts.isEmpty(),
+        "Cannot determine catalog and Identifier from empty name parts");
```

Comment:
"name parts" is an internal thing. What about dropping "parts" and just referring to it as "name"?

Also, no need to capitalize identifier.

Reply:
agreed, done